### PR TITLE
Added funcionality to add a child at a specific index

### PR DESCRIPTION
**Pull request by nestorrente**
I wrote a addChildAt function in displayobject.js and core.js files. I also wrote a function called addAt in displayobject.js, which will be called from core's addChildAt.

Finally, I wrote a removeChildAt function in core.js, in order to maintain the coherence with the Base display object, which already had that functionallity.

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -251,14 +251,32 @@
 			
 			return this;
 		},
-		
+
+		// Method for adding an object to the canvas at a specific index
+		// If index < 0, then then child is added to the beginning of children array
```
Comment (koggdal) [inline src/core.js]: Typo: `then then` => `then the` (in all instances of this comment)

File: src/displayobject.js [hunk 2]
```diff
@@ -351,28 +351,43 @@
...
-					this.core.children.push(this);
+					if (index >= this.core.children.length) {
+						this.core.children.push(this);
+					} else if (index < 0) {
+						this.core.children.unshift(this);
+					} else {
+						this.core.children.splice(index, 0, this);
+					}
+
+					// this.core.children.push(this);
```
Comment (koggdal) [inline src/displayobject.js]: This was probably forgotten here :)

**Comment by nestorrente** [comment]
Hi! You're welcome :grinning: I like your library so much, I'm glad to contribute!

I have pushed changes into my fork. I don't know if this PR have been updated automatically or if I have to do something more :sweat_smile:

**Comment by koggdal** [comment]
Released! http://ocanvas.org/download#v2.9.0

**Pull request merged**